core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala

```diff
@@ -380,20 +381,36 @@ trait WorkflowSubmission extends FutureSupport with LazyLogging with MethodWiths
       case err: JsValue  => AttributeString(err.toString)
     }
 
-  private def resolveDrsUriServiceAccounts(drsUris: Set[String], userInfo: UserInfo)(implicit
+  def resolveDrsSignedUrls(drsUris: Set[String], userInfo: UserInfo)(implicit
     executionContext: ExecutionContext
-  ): Future[Set[String]] =
+  ): Future[Set[String]] = {
+
+    val urisByProvider: Map[Option[String], List[String]] = drsUris.toList.groupBy(DrsResolver.getProvider)
+
+    if (urisByProvider.contains(None)) {
+      logger.warn(s"Some URIs were unparsable: ${urisByProvider(None)}") // TODO What should be logged vs. thrown?
+      throw new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.NotFound, "Unable to parse URI"))
+    }
+
+    val urisToParse = urisByProvider.values.map(_.head).toList
+
     Future
-      .traverse(drsUris) { drsUri =>
-        drsResolver.drsServiceAccountEmail(drsUri, userInfo)
+      .traverse(urisToParse) { drsUri =>
+        drsResolver.drsSignedUrl(drsUri, userInfo)
       }
-      .map { emails =>
-        val collected = emails.collect { case Some(email) =>
-          email
+      .map { urls =>
+        val collected = urls.collect { case Some(url) => url }.toSet
+        if (collected.size != urls.size) {
+          throw new RawlsExceptionWithErrorReport(errorReport =
+            ErrorReport(StatusCodes.InternalServerError,
+                        "One or more URLs could not be resolved"
```

Comment:
I would include in the error message which provider caused the failure so the user might figure out which provider they need to link accounts with